wordle-web/pages/index.vue

```diff
@@ -1,72 +1,108 @@
-      </v-btn>
-    </v-card>
+  <v-container> 
+      <v-card class="text-center">
+          <v-alert v-if="game.gameState != GameState.Playing"
+                   :color="game.gameState == GameState.Won ? 'success' : 'error'"
+                   class="mb-5"
+                   tile>
+              <h3>
+                  You've
+                  {{ game.gameState == GameState.Won ? "Won! 🥳" : "Lost... 😭" }}
+              </h3>
+              <v-card-text>
+                  The word was: <strong>{{ game.secretWord }}</strong>
+              </v-card-text>
+              <v-btn variant="outlined" @click="game.startNewGame()">
+                  <v-icon size="large" class="mr-2"> mdi-restart </v-icon> Restart Game
+              </v-btn>
+          </v-alert>
+          <v-card-title v-else>Wordle</v-card-title>
+
+          <GameBoardGuess v-for="(guess, i) of game.guesses"
+                          :key="i"
+                          :guess="guess" />
+
+          <div class="my-10">
+              <Keyboard />
+          </div>
+
+          <v-btn @click="game.submitGuess()" class="mb-5" color="primary">
+              Guess!
+          </v-btn>
+
+          <v-btn @click="
+          $router.push('/leaderboard');"
+                 class="mb-5 mx-5"
+                 color="primary">
+              Leaderboard
+          </v-btn>
+      </v-card>
   </v-container>
 </template>
 
 <script setup lang="ts">
+import { ref } from 'vue';
 import { Game, GameState } from "../scripts/game";
 import Axios from "axios";
 
-const game: Ref<Game> = ref(new Game(""));
+
+
+
+
+const game: Ref<Game> = ref(new Game("GAMES"));
+
+watch(
+  () => game.value?.gameState,
+  (stateChange) => {
+    switch (stateChange) {
+      case GameState.Won:
+        postScore();
+        break;
+    }
+  }
+);
+
 // Get random word from word list
-getWordFromApi().then((word) => {
+onMounted(() => {
+  getWordFromApi().then((word) => {
   game.value = new Game(word);
+  });
+  window.addEventListener("keyup", onKeyup);
 });
 
 provide("GAME", game.value);
 
 const myGuess = ref("");
 
-onMounted(() => {
-  window.addEventListener("keyup", onKeyup);
-});
-
 onUnmounted(() => {
   window.removeEventListener("keyup", onKeyup);
 });
 
 async function getWordFromApi(): Promise<string> {
-  let wordUrl = "https://wordleapiewu.azurewebsites.net/word";
+  let wordUrl = "word/wordOfTheDay";
 
   const response = await Axios.get(wordUrl);
   console.log("Response from API: " + response.data);
   return response.data;
 }
 
+async function postScore(){
+  let attempts = 0;
+  if(game.gameState == GameState.Won){
+    attempts = game.guessIndex + 1;
+  }
+  else {
+    attempts = game.maxAttempts;
+  }
+  const requestBody = {
+    name: nuxtStorage.localStorage.get("userName", userName.value),
+    gameCount: 1, //since it posts after everygame
+    averageAttempts: attempts,
+    AverageSecondsPerGame: 0
+  };
+  axios.post("http://localhost:5297/Leaderboard", requestBody);
```

Comment:
When this app is deployed to Azure, it tries to communicate to 127.0.0.1:5297, but there is nothing there. This needs to target your API endpoint on Azure to save scores when you aren't running your API locally.